api.js

```diff
@@ -8,50 +8,108 @@ import {
   validateMultiAuthorBatch as validateMultiAuthorBatchWasm,
 } from "./pkg/ssb_validate2_rsjs_wasm.js";
 
-const verifySignatures = (msgs) => {
+// "The buffer module from node.js, for the browser"
+const Buffer = require('buffer/').Buffer;
+
+const stringify = (msg) => JSON.stringify(msg, null, 2);
+
+const verifySignatures = (hmacKey, msgs) => {
   if (!Array.isArray(msgs)) return "input must be an array of message objects";
-  const jsonMsgs = msgs.map((msg) => {
-    return JSON.stringify(msg, null, 2);
-  });
-  return verifySignaturesWasm(jsonMsgs);
+  const jsonMsgs = msgs.map(stringify);
+  let hmacVal;
+  if (!hmacKey) {
+    hmacVal = null;
+  } else {
+    hmacVal = Buffer.isBuffer(hmacKey)
+      ? hmacKey
+      : Buffer.from(hmacKey, "base64");
+    if (typeof hmacKey === "string") {
+      if (hmacVal.toString("base64") !== hmacKey)
+        return "hmac key invalid: string must be base64 encoded";
+    }
+  }
```

Comment:
Sinces lines 20–30 are copy-pasted 5x in other functions, they could be refactored as a function called `toBuffer` (or similar name) which takes the `hmacKey` and returns the `hmacVal`